auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordProgressState.java

```diff
@@ -0,0 +1,44 @@
+package com.firebase.ui.auth.ui.email;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.firebase.ui.auth.data.model.ProgressState;
+
+public final class RecoverPasswordProgressState extends ProgressState {
+    private final String mEmail;
+
+    public RecoverPasswordProgressState(boolean done) {
+        this(done, null, null);
+    }
+
+    public RecoverPasswordProgressState(boolean done,
+                                        @Nullable String email,
+                                        @Nullable Exception e) {
+        super(done, e);
+        mEmail = email;
+
+        if (isSuccessful() && TextUtils.isEmpty(email)) {
+            throw new IllegalStateException("Email cannot be null in the success state");
```

Comment:
nit: Why not expose two constructors?  One that takes a `NonNull` email and represents success and another that takes an `Exception` and represents failure?

Reply:
I was tired? 😆 Yep, done.